toolchains/workspace-pnpm/macros.bzl

```diff
@@ -1021,13 +1021,15 @@ if [ "$install_node_modules" = "True" ]; then
 fi
 exec pnpm run --report-summary "$npm_run_command" -- "${@:4}"
 """, is_executable = True)
-    args = cmd_args([script, str(ctx.attrs.local_node_modules), ctx.label.package, ctx.attrs.command])
+    package_dir = ctx.attrs.package_dir or ctx.label.package
+    args = cmd_args([script, str(ctx.attrs.local_node_modules), package_dir, ctx.attrs.command])
     args.hidden([ctx.attrs.deps])
     args.hidden([ctx.attrs.srcs])
     return [DefaultInfo(), RunInfo(args = args)]
 
 dev_pnpm_task_binary = rule(impl = pnpm_task_binary_impl, attrs = {
     "command": attrs.string(doc = """pnpm command to run"""),
+    "package_dir": attrs.string(default = "", doc = """package dir to run commands from"""),
```

Comment:
we should not expose package_dir IMO.

Rather we should define the target in the correct package (ie core/api) and then alias it from where we want to call it (ie. core/api-ws-server)

Reply:
Fixed